### PR TITLE
rustc_codegen_ssa: cleanup `AtomicOrdering`

### PR DESCRIPTION
**Pull request by tmiasko**
* Remove unused `NotAtomic` ordering.
* Rename `Monotonic` to `Relaxed` - a Rust specific name.
* Derive copy and clone.

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/llvm/ffi.rs [hunk 1]
```diff
@@ -371,7 +371,6 @@ pub enum AtomicOrdering {
     NotAtomic = 0,
     Unordered = 1,
     Monotonic = 2,
-    // Consume = 3,  // Not specified yet.
```
Comment (luqmana) [inline compiler/rustc_codegen_llvm/src/llvm/ffi.rs:374]: Why not keep this comment? Makes it easy to know why there's a random hole at a glance

**Comment by michaelwoerister** [comment]
To understand what's going on here: On the Rust side we use the term "relaxed" where LLVM uses the term "monotonic" and this PR makes the Rust side of things consistently use "relaxed"?

**Comment by tmiasko** [comment]
> To understand what's going on here: On the Rust side we use the term "relaxed" where LLVM uses the term "monotonic" and this PR makes the Rust side of things consistently use "relaxed"?

Yes, that's right.

**Comment by michaelwoerister** [comment]
OK, that makes sense to me 👍
Thanks, @tmiasko!
@bors r+

**Pull request merged**